### ISSUE TITLE
Some Group-Entities are in state "unknown"

### ISSUE THREAD

**Issue by ProfDrYoMan** [SukramJ/hahomematic#552]
**custom_component/hahomematic version (if applicable):**
1.12.4

**Home Assistant version (if applicable):**
2022.7.0

**CCU version:**
- HA-Addon: RaspimaticCCU 3.65.6.20220723 

**Problem-relevant configuration:**

Do you use tls? No
Do you use callback? No
Do you use username and password? Yes
Which interfaces do you use (hmip/bidcos/wired)? hmip/bidcos/groups

**Describe the bug**

After update of RapsimaiticCCU or customHA to latest the groups are not showing up in HA.

Ok in Raspimatic
![image](https://user-images.githubusercontent.com/510353/180818306-7eefe958-55a7-406c-8128-b2b58c8da5dc.png)

Also complete ok and controllable
![image](https://user-images.githubusercontent.com/510353/180818400-02583ec6-523a-4ac4-b838-0d2cb7f78fdf.png)

But in HA:
![image](https://user-images.githubusercontent.com/510353/180818519-ff96e4fb-176f-47e6-8e87-5ca5ec2ffc41.png)

Reboots of both sides (RaspimaticAddon and HA did not help).

**Comment by ProfDrYoMan** [comment]
Just to be completely sure: I did a virtual shutdown/restart of the complete HA (with including raspimaticAddon docker in HA) and now the stuff get's populated again.

If there is no question back, please feel free to close this one.

**Comment by SukramJ** [comment]
I think this relates to [this](https://github.com/danielperna84/custom_homematic#noteworthy-about-entity-states). Playing with the device in in CCU UI should also help.

**Comment by ProfDrYoMan** [comment]
Nope. Playing with the device in CCU did not help. Did that. And even acted directly on the thermostats. Did not bring the entities to life in HA.

**Comment by dfigus** [comment]
I had the same experience yesterday evening when I updated the custom component and HA 
(both to the latest version). After HA restart the groups were restored with state unknown. Note: I did **not** restart the CCU. As I had no time to troubleshoot/debug I reverted back to version 1.11.0. Once I have more time I will update the custom component again, collect all logs during the HA restart and update the issue.

**Comment by SukramJ** [comment]
No, please create a new issue with you own data.

**Comment by SukramJ** [comment]
From my personal experience i can say that the xmlrpc server for groups is one of the worse pieces of software.  I always had the problem that group climate entities were no longer updated after a few days, although the device-based entities were always correct.  the only thing that helped was a restart of the ccu.  the problem has always existed with integration.  With the newer versions, default values ​​are no longer used, instead only the actual values ​​are displayed. The last values ​​can then be restored on subsequent starts.  but this also means that unknown will be displayed if values ​​cannot be loaded, or be sent.  

I've been doing without the groups for a while, now I'm very happy with it.

**Comment by SukramJ** [comment]
@dfigus An alternate view to the data(Screenshot), like with homematic- manager, will be helpful.
From the group device and the included thermostats.

**Comment by dfigus** [comment]
> I always had the problem that group climate entities were no longer updated after a few days, although the device-based entities were always correct. the only thing that helped was a restart of the ccu. 

I don’t habe this experience. I usually update HA every 3-4 weeks and the CCU even less. I never had any issues with groups that were no longer updated (I only use the climate groups in HA). With 1.11.0 all is running fine again.

I will create a new issue with my data as soon as I find some time. This might take some days.

**Comment by ProfDrYoMan** [comment]
Now I have the same issue as SukramJ, groups do not update in HA, but in raspimatic they do.
I never experienced that before, but with latest update of raspimatic and hahomematic (too bad in parallel).

**Comment by ProfDrYoMan** [comment]
@dfigus your updated info would be nice. Did you also update the raspimatic/CCU or did you only update the hahomematic part? If the latter, than it is clearly something wrong in only hahomematic.

**Comment by SukramJ** [comment]
Please check if your CCU is sending events for virtualdevices . You can use Homematic-Manager for this.

**Comment by ProfDrYoMan** [comment]
I expect the need to enable the export of a port in the HA raspimatic addon to check that from another machine via Homematic-Manager. In standard config it will be not exported/available.

Which port number(s) do I need to expose to allow full function of Homematic-Manager?

**Comment by SukramJ** [comment]
80, 2001, 2010,9292

**Comment by ProfDrYoMan** [comment]
If I check quickly with PocketControl (iOS) (I am remote right now) I don't see any updates on the virtual devices. So this seems an issue with Raspimatic.

**Comment by SukramJ** [comment]
@baxxy13 Any idea how to check this in RM?

**Comment by ProfDrYoMan** [comment]
https://homematic-forum.de/forum/viewtopic.php?f=65&t=74912&p=726649#p726649

**Comment by Baxxy13** [comment]
Hmm, i'm actually not up-to-date with RM and the Integration because of holidays. 
The only thing i know that prevents virtual devices from updating are the "zombie-inits". 
But they should not be there after a fresh restart of RM. 
My best tipp actually is to disable ALL! external software (ioBroker, NodeRed, HomemicIP(local), Homematic-Manager) and then reboot RM. Check periodically if the virtual-devices are updated. If (after a day) everything is fine, the Problem comes not from RM.

**Comment by ProfDrYoMan** [comment]
I will anyhow recheck with Homematic Manager at home to ensure that there was no problem being remote in a contained work environment checking over wireguard.

**Comment by Baxxy13** [comment]
As i said, the best check is the WebUI. 
Just compare e.g. Temperature+"Last changed Timestamp" from your virtual device with your wall or radiator thermostat.
They should be (almost) synchronous. 
Homematic-Manager can be part of the "zombie-init" problem because sometimes the de-init fails when closing Homematic-Manager.

My Live-System (no external software access) was updated 3 days ago to latest RM an shows no problems in virtual devices.

**Comment by SukramJ** [comment]
If I understand correctly, then the VirtualDevices server does not clean up orphaned inits like the other servers do if the de-init didn't work.
Can you use a script to determine which zombie inits belong to an IP?
Request: my IP `1.2.3.4` port used is `45634`
Answer: the following ports are still used: `53456`, `44434`

You could then use de-init to remove these inits, and only one init belonging to your own IP address would be active.

**Comment by egnerfl** [comment]
I had the same issue, rolling back to `1.12.0` fixed it for now.

**Comment by SukramJ** [comment]
@egnerfl  This is not helpful. What to do with the information?
From version did you roll back? Did you use 1.12.4 or a newer one?
What is your issue?

**Comment by egnerfl** [comment]
@SukramJ the issue had to be introduced between `1.12.0` and `1.12.5`. I'll try out the in-between versions soon but I guess for everyone who right now faces the issue it is helpful to know that rolling back fixes it for now.

The issue on my side is the exact same @ProfDrYoMan faces, "Some Group-Entities missing in HA". All other devices are synced properly.

**Comment by SukramJ** [comment]
@egnerfl  best option would be to open a new issue, even if it looks like a duplicate to this one,  where you could add all the the information that are requested in the issue template, incl. screenshots, logs, diagnostics.
Thanks.

**Comment by SukramJ** [comment]
@ProfDrYoMan Please retest with 1.12.5 . This should fix the initial unknown state of the group entity.
The other problem with missing updates from the CCU, possibly due to zombie-inits, is not solved by this version

**Comment by ProfDrYoMan** [comment]
1.12.5 is installed since release. I did not see missing group values, but cannot say for sure that the problem is gone.

**Comment by SukramJ** [comment]
OK, so lets close this for now. 
Please come back, if the problem re-occures again.

**Comment by dfigus** [comment]
Sorry for reporting back that late. We have a small baby at home and time is somehow limited right now... 

@ProfDrYoMan & @SukramJ: I can confirm that the problem is solved with 1.12.5. After a HA reboot the groups are no longer unknown. I also checked with homematic manager and events are send (the WebUI is also showing the values correctly).

> @dfigus your updated info would be nice. Did you also update the raspimatic/CCU or did you only update the hahomematic part? If the latter, than it is clearly something wrong in only hahomematic.

I just restarted HA, my CCU is running for more than 2 months and I never had "hanging" group values in the last 2-3 years with similar long CCU uptimes. I restart the CCU rarely (usually only to install a new CCU version or patch the PI).
